src/python/pants/backend/shell/target_types.py

```diff
@@ -268,12 +269,77 @@ class ShellCommandOutputsField(StringSequenceField):
         Use a trailing slash on directory names, i.e. `my_dir/`.
         """
     )
+    removal_hint = "To fix, use `output_files` and `output_directories` instead."
+    removal_version = "2.17.0.dev0"
+
+
+class ShellCommandOutputFilesField(StringSequenceField):
+    alias = "output_files"
+    required = False
+    default = ()
+    help = softwrap(
+        """
+        Specify the shell command's output files to capture.
+
+        For directories, use `output_directories`. At least one of `output_files` and
+        `output_directories` must be specified.
+        """
+    )
+
+
+class ShellCommandOutputDirectoriesField(StringSequenceField):
+    alias = "output_directories"
+    help = softwrap(
+        """
+        Specify full directories of output from the shell command to capture.
+
+        For files, use `output_files`. At least one of `output_files` and
+        `output_directories` must be specified.
+        """
+    )
 
 
 class ShellCommandDependenciesField(ShellDependenciesField):
     supports_transitive_excludes = True
 
 
+class ShellCommandUseOutputDependenciesInRuntimeField(BoolField):
+    alias = "use_dependencies_at_runtime"
+    default = True
+    help = softwrap(
+        """
+        Allows you to separate dependencies between those required to run the shell command and
+        those required to successfully use the output artifacts.
+
+        If `True`, the dependencies specified in the `dependencies` field will be materialized
+        into the runtime sandbox (with any package dependencies built), and also included in any
+        transitive dependency resolution involving this target. Dependencies specified in
+        `runtime_dependencies` will also be materialized into the sandbox.
+
+        If `False`, the dependencies in the `dependencies` field will only be used for transitive
+        dependency resolution, and only the `runtime_dependencies` field will be materialized into
+        the sandbox.
+        """
+    )
+
+
+class ShellCommandRuntimeDependenciesField(SpecialCasedDependencies):
+    alias = "runtime_dependencies"
+    help = softwrap(
+        """
+        The runtime dependencies for this shell command.
+
+        Dependencies specified here are those required to make the command complete successfully
+        (e.g. file inputs, binaries compiled from other targets, etc), but NOT required to make
+        the output side-effects useful. Dependencies that are required to use the side-effects
+        produced by this command should be specified using the `dependencies` field.
+
+        If you have fully separated your dependencies between runtime and output dependencies,
+        you can set `use_dependencies_at_runtime = False`.
+        """
+    )
```

Comment:
I don't quite understand these fields. I think there's some confusion here between when "runtime" is. between two readers.

To me, there's "build time" dependencies (those require for this command to "build", or more accurately execute). Then there's runtime dependencies (dependencies the thing this produces needs when it is run).

In that wording of it, this is flipped. And I suggest we rename the vanilla `dependencies` field to be more specific as well.

---

Additionally I think the toggle of `use_dependencies_at_runtime` is a bit much. If something needs to be in both I don't think it's too painful to specify it in both (considering how rare that might be?)

Reply:
@thejcannon I think renaming `dependencies` to `output_dependencies` using the deprecation alias makes sense -- I'll do that.

`use_dependencies_at_runtime` is there so as not to break existing `experimental_shell_commands`. We can either get rid of it entirely and break existing installations, or go through a deprecation cycle. Normally I would be more gung-ho about changing experimental APIs, but we've pointed a _lot_ of people at esc lately.

Reply:
I should add, I'm kinda looking at this using the lens of this being a precedent for broader changes throughout Pants.

Reply:
I'm trying not to look too far into the broader context, this is an `experimental` API, where we're comfortable with deciding that things are a bad idea after a version or two

It's only a precedent if it ends up being good ;)

Reply:
At the very least, can we rename from `runtime_dependencies` to `execution_dependencies`?

Reply:
yes, that's coming up right now

Reply:
it takes a long time to fix things that are inside string literals

Reply:
I was also a bit confused about the 'runtime' here, for similar reasons.

Here's an alternative proposal:

1. `dependencies` stays as is, and is tightened to be just used to execute the shell command (i.e. what's called `runtime_dependencies` in the current PR)
2. a _new_ `output_dependencies` field is added
3. the `use_dependencies_at_runtime` compatibility field is renamed to `use_dependencies_for_output_too` (or something)

But I don't have a good sense of the cost/benefits of using the conventional name for the 'common' case (I imagine usually dependencies will only be used for running the shell command itself, it is for us) vs. something theoretically clearer (`build_dependencies`, `codegen_dependencies`, `execution_dependencies`, `tool_dependencies`, `dependencies_just_used_to_run_command`, ... I cannot think of a very good one).

Reply:
yeah, that sounds like a better solution. Doing that.

Reply:
It tosses deprecation to the wind. But `experimental` is an amazing scapegoat

Reply:
It does. My only hesitation is that we've sent a _lot_ of people to `esc` recently, so I'm trying to be cautious about changes for the time being

Reply:
being experimental, we could have 1 .dev release with some deprecated field names juggling to have `./pants update-build-files` fix renames for them, then in the next .dev we do it again, landing on the desired end result. Those that care will just have to lock in to these dev versions to auto-fix their esc field names.

Reply:
In this case I don't think we can autofix anything as people need to decide where their deps go 🤷‍♂️